editor/modes/visual-editor/block.js

```diff
@@ -133,6 +134,10 @@ class VisualEditorBlock extends wp.element.Component {
 
 		const { onSelect, onStartTyping, onHover, onMouseLeave, onFocus, onInsertAfter } = this.props;
 
+		const toolbars = settings.controls && settings.controls.length && ! Array.isArray( first( settings.controls ) )
```

Comment:
I'm not sure I understand the magic here :P but it seems that if we have mixed values, the controls outside arrays are included in the previous array, which I think is not the desired behaviour if we ever have mixed controls. I'd think these "orphans" should be grouped together.

Yes. I'll drop the `first`, it seems enough for now to have only Arrays or only orphans